test/edu/psu/sweng500/team8/gui/SudokuGUITest.java

```diff
@@ -310,4 +310,106 @@ public void testSudokuGUI_PencilMarkMode_EnteringNumber_EmptyCell() {
 		Assert.assertEquals("", ((JLabel)selectedCellGUI.getpPencilMarkDisplayCell().getComponent(8)).getText());
 		
 	}
+	
+	@Test
+	public void testSudokuGUI_GameChangedStatus()
+	{
+		SudokuGUI sudokuGUI = new SudokuGUI();
+		JPanel mainJPanel = ((JPanel)((JLayeredPane)((JRootPane) sudokuGUI.getComponents()[0]).getComponents()[1]).getComponents()[0]);
+		
+		JButton newGameButton = (JButton)mainJPanel.getComponents()[18];
+		newGameButton.doClick();
+		
+		BoardGUI boardGUI = (BoardGUI)mainJPanel.getComponents()[0];
+
+		Assert.assertEquals(sudokuGUI.isGameChanged(), false);
+		
+		/* Find Empty Cell */
+		Cell emptyCell = null;
+		for (int i = 0; i < 9; i++) {
+			for (int j = 0; j < 9; j++) {
+				Cell currentCell = sudokuGUI.getGameSession().getGameBoard().getCellGrid().getCell(i, j);

```

Comment:
(Performance) Cache the cellGrid outside the loop rather than querying for it 81 times